chroma-core/PyTorch-YOLOv3
# M1 support

This PR adds M1 support to the YOLO model. This provides an approximately 10x speedup over CPU-only in the model inference pass, with some mild performance losses at non-maximum suppression. 

# Test
- Install M1 torch in your virtualenv. `pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cpu`
- Install YOLOv3 dependencies: `pip install .`
- Do an inference run with mps enabled. Note that because not all torch operators have been implemented for M1 yet, it's necessary to set an environment variable. `PYTORCH_ENABLE_MPS_FALLBACK=1 python pytorchyolo/detect.py --use_mps `
